authentication/mtls.go

```diff
@@ -13,65 +13,65 @@ type MTLSConfig struct {
 	CAs    []*x509.Certificate
 }
 
-// NewMTLS creates a set of Middlewares for all specified tenants.
-func NewMTLS(configs []MTLSConfig) map[string]Middleware {
-	middlewares := map[string]Middleware{}
+// NewMTLS create Middleware for a tenant.
+func NewMTLS(config MTLSConfig) Middleware {
+	middleware := func(next http.Handler) http.Handler {
+		caPool := x509.NewCertPool()
 
-	for _, c := range configs {
-		c := c
-		middlewares[c.Tenant] = func(next http.Handler) http.Handler {
-			caPool := x509.NewCertPool()
+		if len(config.CAs) == 0 {
```

Comment:
This is surprising - it will trigger nil panics. I wonder if we should not - either panic `NewMTLS` function since it's a programmatic error potentially or return error if we want to handle that on user side. WDYT?

Reply:
Also, I wonder - we don't want to take local host trusted CA in default case? 🤔

Reply:
ping

Reply:
good catch :) right, this check was not required. removed this check.